reframe/frontend/loader.py

```diff
@@ -79,7 +81,12 @@ def _validate_source(self, filename):
         further tests and finalizes and validation."""
 
         with open(filename, 'r') as f:
-            source_tree = ast.parse(f.read())
+            try:
+                source_tree = ast.parse(f.read())
+            except SyntaxError as e:
+                message = ('\nRegression check syntax error:\n{0}Line: {1}\n'
+                           'File: {2}'.format(e.text, e.lineno, filename))
+                raise ReframeSyntaxError(message)
```

Comment:
I don't like very much reraising a real `SyntaxError` as `ReframeSyntaxError`. The purpose of the latter is just to denote errors in the usage of the decorators. I agree the name is not very accurate, but I couldn't come up with a better name. 

The problem here is firstly conceptual: `ReframeSyntaxError`s *are* `ReframeError`s, but a real `SyntaxError` can never be a `ReframeError`. Secondly, reraising it differently may hide useful information from real problems.

The solution to this problem is really simple! Just pass a `filename` argument to `parse()`. Have a look [here](https://docs.python.org/3.6/library/ast.html#ast.parse).